static/app/views/dashboards/widgetCard/widgetCardContextMenu.tsx

```diff
@@ -352,13 +356,14 @@ export function getMenuOptions(
       key: 'duplicate-widget',
       label: t('Duplicate Widget'),
       onAction: () => onDuplicate?.(),
-      disabled: widgetLimitReached,
+      disabled: widgetLimitReached || !hasEditAccess,
     });
 
     menuOptions.push({
       key: 'edit-widget',
       label: t('Edit Widget'),
       onAction: () => onEdit?.(),
+      disabled: !hasEditAccess,
```

Comment:
There is also a button in the widget viewer (which opens up when you click the button next to the `...` context menu) that has an edit widget button. Could we also disable that? Doesn't need to be this PR.